akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1125,7 +1128,14 @@ private[stream] final class RecoverWith[T, M](pf: PartialFunction[Throwable, Gra
       override def onPull(): Unit = pull(in)
     })
 
-    def onFailure(ex: Throwable) = if (pf.isDefinedAt(ex)) switchTo(pf(ex)) else failStage(ex)
+    def onFailure(ex: Throwable) =
+      if ((maximumRetries == InfiniteRetries || attempt < maximumRetries) && pf.isDefinedAt(ex)) {
+        switchTo(pf(ex))
+      }
+      else {
```

Comment:
Do you have a style guide somewhere? Should I remove the braces around the if {} even if the statement is on the next line? Or should I put the statement on the same line even though it's going to be longer than 120 chars?